hudi-common/src/main/java/org/apache/hudi/common/table/timeline/HoodieInstant.java

```diff
@@ -30,7 +34,23 @@
  *
  * @see HoodieTimeline
  */
-public class HoodieInstant implements Serializable {
+public class HoodieInstant implements Serializable, Comparable<HoodieInstant> {
+
+  /**
+   * A COMPACTION action eventually becomes COMMIT when completed. So, when grouping instants
+   * for state transitions, this needs to be taken into account
+   */
+  private static final Set<String> comparableActions = Arrays.stream(new String[] { HoodieTimeline.COMMIT_ACTION,
```

Comment:
Hmm . seems most of the issue.stems from this switching of compaction => commit? Just throwing out an idea to see if we can just call talk about compaction as an implementation, but have the action be just commit? i.e remove Compaction action and replace with Commit given we have requested and inflight there? would that simplify the design? does it open new migration pains?

Reply:
Added a jira to track this : https://jira.apache.org/jira/browse/HUDI-384